packages/playground/src/weblets/tf_staticwebsite.vue

```diff
@@ -0,0 +1,264 @@
+  ) {
+    domain.value = selectionDetails.value?.domain?.enabledCustomDomain
+      ? selectionDetails.value.domain.customDomain
+      : subdomain + "." + selectionDetails.value?.domain?.selectedDomain?.publicConfig.domain;
+  }
+
+  let grid: GridClient | null;
+  let vm: any;
+
+  try {
+    layout.value?.validateSSH();
+    grid = await getGrid(profileManager.profile!, projectName);
+
+    await layout.value.validateBalance(grid!);
+
+    vm = await deployVM(grid!, {
+      name: name.value,
+      network: {
+        addAccess: selectionDetails.value!.domain!.enableSelectedDomain,
+        accessNodeId: selectionDetails.value?.domain?.selectedDomain?.nodeId,
+      },
+      machines: [
+        {
+          name: name.value,
+          cpu: solution.value.cpu,
+          memory: solution.value.memory,
+          disks: [
+            {
+              size: solution.value.disk,
+              mountPoint: "/var/lib/docker",
+            },
+          ],
+          flist: flist.value,
+          entryPoint: flist.entryPoint,
+          publicIpv4: ipv4.value,
+          mycelium: mycelium.value,
+          envs: [
+            { key: "SSH_KEY", value: selectedSSHKeys.value },
+            { key: "GITHUB_URL", value: githubUrl.value },
+            { key: "GITHUB_BRANCH", value: githubBranch.value },
+            { key: "HTML_DIR", value: root.value ? "website/" + root.value : "website" },
+            { key: "USER_DOMAIN", value: selectionDetails.value?.domain?.enabledCustomDomain ? domain.value : "" },
+            { key: "STATICWEBSITE_DOMAIN", value: domain.value },
+          ],
+          nodeId: selectionDetails.value!.node!.nodeId,
+          rentedBy: dedicated.value ? grid!.twinId : undefined,
+          certified: certified.value,
+          rootFilesystemSize: rootFilesystemSize.value,
+        },
+      ],
+    });
+  } catch (e) {
+    return layout.value.setStatus("failed", normalizeError(e, "Failed to deploy a Static Website instance."));
+  }
+  if (domain.value) {
+    if (!selectionDetails.value?.domain?.enableSelectedDomain) {
+      vm[0].customDomain = selectionDetails.value?.domain?.customDomain;
+      return;
+    }
+
+    try {
+      layout.value.setStatus("deploy", "Preparing to deploy gateway...");
+
+      await deployGatewayName(grid, selectionDetails.value.domain, {
+        subdomain,
+        ip: vm[0].interfaces[0].ip,
+        port: 9000,
+        network: vm[0].interfaces[0].network,
+      });
+      finalize(vm);
+    } catch (e) {
+      layout.value.setStatus("deploy", "Rollbacking back due to fail to deploy gateway...");
+
+      await rollbackDeployment(grid!, name.value);
+      layout.value.setStatus("failed", normalizeError(e, "Failed to deploy a Static Website instance."));
+    }
+  }
+  finalize(vm);
+}
+
+async function isGithubRepoExist(githubUrl: string) {
+  try {
+    const username = githubUrl.substring("https://github.com/".length, githubUrl.lastIndexOf("/"));
```

Comment:
you can extract username and the repo name using a regex, but still what you can do is

https://github.com/xmonader/repo
replace https://github.com with ""
replace leading / with ""
replace .git with ""
split on "/" to get the username and the repo